felix/statusrep/status_file_reporter.go

```diff
@@ -239,18 +240,12 @@ func (fr *EndpointStatusFileReporter) handleEndpointUpdate(e interface{}) {
 	switch m := e.(type) {
 	case *proto.WorkloadEndpointStatusUpdate:
 		key := names.WorkloadEndpointIDToWorkloadEndpointKey(m.Id, fr.hostname)
-		if key == nil {
-			logrus.WithField("update", e).Warn("Couldn't generate a WorkloadEndpointKey from update")
-			return
-		}
-		fr.statusDirDeltaTracker.Desired().Add(names.WorkloadEndpointKeyToStatusFilename(key))
+		fn := names.WorkloadEndpointKeyToStatusFilename(key)
+		fr.statusDirDeltaTracker.Desired().Add(fn)
```

Comment:
Removed the nil check here?  Is that no longer possible?

Reply:
Will replace with a nil check on the original update's ID since that's the only case where the util func will return nil